src/cpp/session/SessionModuleContext.cpp

```diff
@@ -141,15 +142,25 @@ class ConsoleInputService : boost::noncopyable
 private:
    
    void run()
+   {
+      try
+      {
+         runImpl();
```

Comment:
Just curious, what is gained by moving this out to a separate Impl method? Usually see that pattern used to help expose things for unit testing, etc.

Reply:
Does reduce nesting so readability is also a valid answer.